openedx/core/djangoapps/oauth_dispatch/models.py

```diff
@@ -43,3 +46,88 @@ def verify_access_token_as_expired(cls, access_token):
         is set at the beginning of the epoch which is Jan. 1, 1970
         """
         return access_token.expires == datetime(1970, 1, 1, tzinfo=utc)
+
+
+class ScopedApplication(AbstractApplication):
+    """
+    Custom Django OAuth Toolkit Application model that enables the definition
+    of scopes that are authorized for the given Application.
+    """
+    FILTER_USER_ME = 'user:me'
+
+    # TODO: Remove the id field once we perform the inital migrations for this model.
+    # We need to copy data over from the oauth2_provider.models.Application model to
+    # this new model with the intial migration and the model IDs will need to match
+    # so that existing AccessTokens will still work when switching over to the new model.
+    # Once we have the data copied over we can move back to an auto-increment primary key.
+    id = models.IntegerField(primary_key=True)
+    scopes = ListCharField(
+        base_field=models.CharField(max_length=32),
+        size=10,
```

Comment:
This also needs to update to 25 (to be consistent with `max_length`.

Reply:
Good catch.